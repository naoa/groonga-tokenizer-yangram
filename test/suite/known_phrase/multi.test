register tokenizers/yangram

table_create known_phrases TABLE_PAT_KEY ShortText
load --table known_phrases
[
{"_key": "【請求項1】"},
{"_key": "【請求項2】"},
{"_key": "【請求項3】"}
]
tokenize TokenYaBigram "【請求項１】検索装置【請求項２】検索システム【請求項３】" NormalizerAuto --mode ADD
tokenize TokenYaBigram "【請求項１】検索装置【請求項２】検索システム【請求項３】" NormalizerAuto --mode GET
