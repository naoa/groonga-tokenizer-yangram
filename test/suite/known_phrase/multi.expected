register tokenizers/yangram
[[0,0.0,0.0],true]
table_create known_phrases TABLE_PAT_KEY ShortText
[[0,0.0,0.0],true]
load --table known_phrases
[
{"_key": "【請求項1】"},
{"_key": "【請求項2】"},
{"_key": "【請求項3】"}
]
[[0,0.0,0.0],3]
tokenize TokenYaBigram "【請求項１】検索装置【請求項２】検索システム【請求項３】" NormalizerAuto --mode ADD
[
  [
    0,
    0.0,
    0.0
  ],
  [
    {
      "value": "【請求項1】",
      "position": 0
    },
    {
      "value": "検索",
      "position": 1
    },
    {
      "value": "索装",
      "position": 2
    },
    {
      "value": "装置",
      "position": 3
    },
    {
      "value": "置",
      "position": 4
    },
    {
      "value": "【請求項2】",
      "position": 5
    },
    {
      "value": "検索",
      "position": 6
    },
    {
      "value": "索シ",
      "position": 7
    },
    {
      "value": "シス",
      "position": 8
    },
    {
      "value": "ステ",
      "position": 9
    },
    {
      "value": "テム",
      "position": 10
    },
    {
      "value": "ム",
      "position": 11
    },
    {
      "value": "【請求項3】",
      "position": 12
    }
  ]
]
tokenize TokenYaBigram "【請求項１】検索装置【請求項２】検索システム【請求項３】" NormalizerAuto --mode GET
[
  [
    0,
    0.0,
    0.0
  ],
  [
    {
      "value": "【請求項1】",
      "position": 0
    },
    {
      "value": "検索",
      "position": 1
    },
    {
      "value": "装置",
      "position": 3
    },
    {
      "value": "【請求項2】",
      "position": 5
    },
    {
      "value": "検索",
      "position": 6
    },
    {
      "value": "シス",
      "position": 8
    },
    {
      "value": "テム",
      "position": 10
    },
    {
      "value": "【請求項3】",
      "position": 12
    }
  ]
]
